vllm/model_executor/layers/spec_decode_base_sampler.py

```diff
@@ -36,9 +36,12 @@ def __init__(self,
         self.num_emitted_tokens: Optional[torch.Tensor] = None
         self.num_draft_tokens: int = 0
 
-    def init_gpu_tensors(self, rank: int) -> None:
+    def init_gpu_tensors(self, device: Union[int, str]) -> None:
```

Comment:
nit: preference to move this to own variable like `Device = Union[int, str]` to prevent bloat.